core/query/src/main/java/datawave/core/query/logic/filtered/DateRangeFilteredQueryLogic.java

```diff
@@ -0,0 +1,50 @@
+package datawave.core.query.logic.filtered;
+
+import java.util.Set;
+
+import org.apache.accumulo.core.client.AccumuloClient;
+import org.apache.accumulo.core.security.Authorizations;
+import org.apache.log4j.Logger;
+
+import datawave.core.query.configuration.GenericQueryConfiguration;
+import datawave.microservice.query.Query;
+
+public class DateRangeFilteredQueryLogic extends FilteredQueryLogic {
+    public static final Logger log = Logger.getLogger(DateRangeFilteredQueryLogic.class);
+
+    @Override
+    public GenericQueryConfiguration initialize(AccumuloClient connection, Query settings, Set<Authorizations> runtimeQueryAuthorizations) throws Exception {
+        if (canRunQuery(settings, runtimeQueryAuthorizations)) {
+            // may need to adjust the Query to run within the defined range
+            if (getFilter() instanceof QueryLogicFilterByDate) {
+                QueryLogicFilterByDate filter = (QueryLogicFilterByDate) getFilter();
+                applyFilterDateRange(settings, filter);
+            } else {
+                log.warn("expected QueryLogicFilterByDate filter, but got: " + getFilter().getClass() + " No date range adjustment will be done");
```

Comment:
Minor bit. Add period  or other separating punctuation after class name before the follow on comment?

Reply:
will rephrase